src/common/com/intellij/plugins/haxe/ide/HaxeImportOptimizer.java

```diff
@@ -69,6 +82,49 @@ private static void optimizeImports(PsiFile file) {
       unusedImportStatement.delete();
     }
 
-    // todo: rearrange
+    // TODO Optimize this method, it takes a lot of time.
+
+    // TODO Remove unused usings.
+  }
+
+  public static void reorderImports(final PsiFile file) {
+    List<HaxeImportStatementRegular> allImports = UsefulPsiTreeUtil.getAllImportStatements(file);
+
+    if (allImports.size() < 2) {
+      return;
+    }
+
+    List<String> sortedImports = new ArrayList<String>();
+
+    for (HaxeImportStatementRegular currentImport : allImports) {
+      sortedImports.add(currentImport.getText());
+    }
+
+    Collections.sort(sortedImports);
+
+    final Document document = PsiDocumentManager.getInstance(file.getProject()).getDocument(file);
+    if (document != null) {
+      final PsiDocumentManager documentManager = PsiDocumentManager.getInstance(file.getProject());
```

Comment:
These three lines will read a lot better if you did (paraphrased):

docMgr = getInstance(file.getProject);
document = docMgr.getDocument(file);
if (document) { ...